acceptance/tests/external_facts/user_uses_default_external_fact_dir.rb

```diff
@@ -0,0 +1,54 @@
+test_name "Non-root user default external facts directory (facts.d) is searched for facts"
+
+require 'facter/acceptance/user_fact_utils'
+extend Facter::Acceptance::UserFactUtils
+
+#
+# This test is intended to ensure that executable external facts placed into the
+# default facts.d directory are properly found and resolved by Facter when run
+# as a user other than root. In such a case, facts.d is found in the user's homedir.
+#
+
+user_name = 'testuser'
+unix_content = <<EOM
+#!/bin/sh
+echo "external_fact=testvalue"
+EOM
+
+win_content = <<EOM
+echo "external_fact=testvalue"
+EOM
+
+agents.each do |agent|
+  if agent['platform'] =~ /windows/
+    content = win_content
+  else
+    content = unix_content
+  end
+
+  step "Agent #{agent}: create new user"
+  on(agent, "useradd #{user_name} --home /home/#{user_name}")
+  on(agent, "mkdir -p /home/#{user_name}")
+  on(agent, "chown #{user_name}:#{user_name} /home/#{user_name}")
+
+  step "Agent #{agent}: setup default user external facts directory (facts.d)"
+  os_version = on(agent, facter('kernelmajversion')).stdout.chomp.to_f
+  factsd = get_non_root_factsd_dir(agent['platform'], os_version, user_name)
+  on(agent, "mkdir -p '#{factsd}'")
+
+  step "Agent #{agent}: create an executable external fact in default facts.d"
+  ext = get_external_fact_script_extension(agent['platform'])
+  ext_fact = File.join(factsd, "external_fact#{ext}")
+  create_remote_file(agent, ext_fact, content)
+  on(agent, "chown #{user_name}:#{user_name} #{ext_fact}")
+  on(agent, "chmod +x #{ext_fact}")
+
+  teardown do
+    on(agent, "rm -f '#{ext_fact}'")
+  end
+
+  step "Agent #{agent}: the external fact should resolve"
+  on(agent, "su - #{user_name} -c 'facter external_fact'") do
```

Comment:
Sadly this won't work on windows as cygwin doesn't support`su`. You could use `runas`, though you'll have to figure out the syntax in cygwin+ssh environment. Also `runas` will require you to specify a password (so you'll need to specify a password when the user account is created, and should likely be randomized each test run).